include/openthread/coap.h

```diff
@@ -367,6 +420,26 @@ typedef struct otCoapResource
     struct otCoapResource *mNext;    ///< The next CoAP resource in the list
 } otCoapResource;
 
+/**
+ * This structure represents a CoAP resource with blockwise transfer.
+ *
+ */
+typedef struct otCoapBlockwiseResource
+{
+    const char *               mUriPath;       ///< The URI Path string
+    otCoapRequestHandler       mHandler;       ///< The callback for handling a received request
+    otCoapBlockwiseReceiveHook mReceiveHook;   /* The callback for handling incoming blockwise transfer.
+                                                  This callback is available when
+                                                  OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+                                                  configuration is enabled. */
+    otCoapBlockwiseTransmitHook mTransmitHook; /* The callback for handling outgoing blockwise transfer.
+                                                  This callback is available when
+                                                  OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
+                                                  configuration is enabled. */
+    void *                          mContext;  ///< Application-specific context
+    struct otCoapBlockwiseResource *mNext;     ///< The next CoAP resource in the list
```

Comment:
Suggest re-arranging the longer comment before the definition and start with `/** ` for doxygen to associate it correctly with the member variable.

```suggestion
    otCoapRequestHandler       mHandler;       ///< The callback for handling a received request

    /** The callback for handling incoming blockwise transfer.
     *  This callback is available when
     *  OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
     *  configuration is enabled.
     */
    otCoapBlockwiseReceiveHook mReceiveHook;

    /** The callback for handling outgoing blockwise transfer.
     *  This callback is available when
     *  OPENTHREAD_CONFIG_COAP_BLOCKWISE_TRANSFER_ENABLE
     *  configuration is enabled. 
     */
    otCoapBlockwiseTransmitHook mTransmitHook; 

    void *                          mContext;  ///< Application-specific context
    struct otCoapBlockwiseResource *mNext;     ///< The next CoAP resource in the list
```